### PR TITLE
Add a CHANGELOG

### PR DESCRIPTION
**Pull request by kylef**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  connecting:
+
+  - Looking up your hostname...
+  - Checking Ident
+  - Found your hostname
+  - Got Ident response
+
+### Bug Fixes
+
+- We now detect when building the module with an unsupported compiler and now
```
Comment (iamDecode) [inline CHANGELOG.md]: This sentence abruptly ends, probably not intended.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Changelog for Palaver ZNC Module
+
+## Master
```
Comment (iamDecode) [inline CHANGELOG.md]: Might be a bit confusing, as master is "also" 1.0.1 according to the readme etc. Maybe make master 1.0.2 :)

Comment (kylef) [inline CHANGELOG.md]: Master contains the changes pending to be released, once we cut a release the future version number will go here.

Master isn't actually 1.0.1, it's 1.0.1 + changes for next release, the number in README shows the last version and not what master is currently.

Make sense?

Comment (iamDecode) [inline CHANGELOG.md]: Yea I get that, but it might be confusing to some ;) Let's keep it master.

File: CHANGELOG.md (hunk 1 above)
Comment (kylef) [inline CHANGELOG.md]: there is continuation on the lines below it?

Comment (iamDecode) [inline CHANGELOG.md]: Oh haha, error as a verb. Maybe make it "return/throw an error" :)

Comment (kylef) [inline CHANGELOG.md]: Sure, I've reworded.

**Pull request merged**